### PR TITLE
[embedded] Add a section on Strings into the User Manual

### PR DESCRIPTION
**Pull request by kubamracek**
Also, update the feature table, and expand docs on allocations.

### CODE REVIEW

**Review by rauhul — APPROVED** [review]
<3 documentation updates

**Comment by kubamracek** [comment]
@swift-ci please smoke test

**Review by phausler — APPROVED** [review]

File: docs/EmbeddedSwift/UserManual.md [hunk 1]
```diff
@@ -98,6 +98,57 @@ Segment __TEXT: 16384
...
+To resolve this, link in the libswiftUnicodeDataTables.a that's in Swift toolchain's resource directory (`lib/swift/`) under the target triple that you're using:
+
+```bash
+$ swiftc <inputs> -target armv6m-none-none-eabi -enable-experimental-feature Embedded -wmo -c -o output.o
+$ ld ... -o binary output.o $(dirname `which swiftc`)/../lib/swift/embedded/armv6m-none-none-eabi/libswiftUnicodeDataTables.a
+```
+
+**Unicode data tables are required for (list not exhaustive):**
+
+- Comparing String objects for equality
```
Comment (karwa) [inline docs/EmbeddedSwift/UserManual.md:138]: Also sorting Strings (since canonically-equivalent strings must sort identically).

**Comment by kubamracek** [comment]
@swift-ci please smoke test

**Pull request merged**